src/Compiler/pars.fsy

```diff
@@ -2709,12 +2710,14 @@ hardwhiteLetBindings:
 /* A 'do ...' statement */
 hardwhiteDoBinding:
   | ODO typedSequentialExprBlock hardwhiteDefnBindingsTerminator
-      { let mDoKwd = rhs parseState 1
-        let bindingSetRange = unionRanges mDoKwd $2.Range
+      { let mDo = rhs parseState 1
+        let _, _, mDone = $3
+        let mEnd = mDone |> Option.defaultWith (fun _ -> $2.Range)
```

Comment:
Is evaluating `$2.Range` expensive? Or could this just be `Option.defaultValue $2.Range`?